kit/Delta.hpp

```diff
@@ -347,9 +348,12 @@ class DeltaGenerator {
                     // TODO: if offsets are >256 - use 16bits?
                     if (lastCopy > 0)
                     {
-                        char cnt = output[lastCopy];
-                        if (output[lastCopy + 1] + cnt == (char)(match) &&
-                            output[lastCopy + 2] + cnt == (char)(y))
+                        // check if we can extend the last copy
+                        uint8_t cnt = output[lastCopy];
+                        if (output[lastCopy + 1] + cnt == match &&
+                            output[lastCopy + 2] + cnt == y &&
+                            // make sure we're not copying from out of bounds of the previous tile
+                            output[lastCopy + 1] + cnt + 1 < prev.getHeight())
```

Comment:
Seems reasonable - I wonder if we have a unit test for this use-case =) would be good to have one, and also one to check that we do generate multi-row copy outputs still.

Reply:
we do - the one I added to DeltaTests tests exactly this - should fail without this change in Delta.hpp and we do generate multi-row copies still, there's further details in the task report.

Reply:
fair enough.